fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/InternalBiomeData.java

```diff
@@ -0,0 +1,183 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.biomes;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.base.Preconditions;
+
+import net.fabricmc.fabric.api.biomes.v1.OverworldClimate;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.Biomes;
+import net.minecraft.world.biome.layer.BiomeLayers;
+
+/**
+ * Lists and maps for internal use only! Stores data that is used by the various mixins into the world generation
+ */
+public final class InternalBiomeData {
+
+	private InternalBiomeData() {
+	}
+
+	private static final EnumMap<OverworldClimate, WeightedBiomePicker> OVERWORLD_BASE_MODDED_BIOME_PICKERS = new EnumMap<>(OverworldClimate.class);
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_HILLS_MAP = new HashMap<>();
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_SHORE_MAP = new HashMap<>();
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_EDGE_MAP = new HashMap<>();
+	private static final Map<Biome, VariantTransformer> OVERWORLD_VARIANT_TRANSFORMERS = new HashMap<>();
+
+	private static final Map<Biome, Biome> OVERWORLD_RIVER_MAP = new HashMap<>();
+	private static final List<Biome> OVERWORLD_INJECTED_BIOMES = new ArrayList<>();
+	private static final Set<Biome> SPAWN_BIOMES = new HashSet<>();
+
+	public static void addOverworldBaseBiome(OverworldClimate climate, Biome biome, double weight) {
+		Preconditions.checkArgument(climate != null && biome != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_BASE_MODDED_BIOME_PICKERS.computeIfAbsent(climate, k -> new WeightedBiomePicker()).addBiome(biome, weight);
+		OVERWORLD_INJECTED_BIOMES.add(biome);
+	}
+
+	public static void addOverworldHillsBiome(Biome parent, Biome hills, double weight) {
+		Preconditions.checkArgument(parent != null && hills != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
```

Comment:
```suggestion
		Preconditions.checkArgument(weight > 0.0, "Weight is less than 0 (%s)", weight);
```

Reply:
it could also be == 0 though, not just less than

Reply:
Updated suggestion

Reply:
`than` should be before the `or`

Reply:
Note to self, don't quick-fix sentences at 4am